### DISCUSSION TITLE
lit reactive lifecycle method invocation sequence

### DISCUSSION THREAD

**Discussion by sameerkulkarnidev** [lit/lit#2547] [Q&A]
Hi All!

I came across a case while using a lit component in a react application, where `updated` callback is triggered after the `disconnectedCallback` is executed. Not sure if its a bug or an expected behavior. This could trigger some unwanted flows after the component is disconnected. It can be handled at component level manually, but it would be nice if `updated` hook itself is not triggered once the component is disconnected.

Here is a [sample link](https://codesandbox.io/s/hungry-cdn-gwodl2) where the issue is reproducible (Please check console output for the logs).

**Comment by sorvell** [comment]
Yeah, this is a tricky issue, but it's an intentional part of the design. 

Since it's possible to use the update lifecycle for tasks not specifically related to rendering or being in the DOM, it's not safe to suspend or discard an update when an element is disconnected. We have experimented with suspending updates while an element is disconnected and completing them on connection, but it led to confusing behavior and introduced other corner cases.

I think avoiding setting the property on the element you're about to discard is the pattern you want to try to adopt. The behavior of React in the linked example is a little confusing. I guess the state changes are not batched because of the use of `setTimeout`. So it's doing 2 separate synchronous renders. If you reverse the property sets or take out the `setTimeout`, then the Lit property is not set and therefore the element is not updated.
